paddle/phi/kernels/fft_grad_kernel.h

```diff
@@ -0,0 +1,48 @@
+// Copyright (c) 2022 PaddlePaddle Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+
+#include <string>
+#include <vector>
+#include "paddle/phi/core/dense_tensor.h"
+
+namespace phi {
+template <typename T, typename Context>
+void FFTC2CGradKernel(const Context& ctx,
+                      const DenseTensor& out_grad,
+                      const std::vector<int64_t>& axes,
+                      const std::string& normalization,
+                      bool forward,
+                      DenseTensor* x_grad);
+
+template <typename T, typename Context>
+void FFTC2RGradKernel(const Context& ctx,
+                      const DenseTensor& out_grad,
+                      const std::vector<int64_t>& axes,
+                      const std::string& normalization,
+                      bool forward,
+                      int64_t last_dim_size,
```

Comment:
According to the previous consideration, this parameter `last_dim_size` is not necessarily required for grad direction, nor is it in the currently submitted national standard. This difference should be recorded, and we can modify it later.

Reply:
Currently last_dim_size is used in infer meta function of fft_c2r, to provide better compile time inference.

Because of the current argument picking mechanism connecting paddle operators and phi infer meta functions and kernel functions, infer meta functions cannot acquire parameters that are not in the parameter list of the kernel function. Thus, these parameters are all included in the parameter list of kernel functions.